ApiClient.ts

```diff
@@ -1,134 +1,203 @@
-import fetch, { FetchError, fileFromSync, FormData } from 'node-fetch';
+import fetch, { FetchError, fileFromSync, FormData } from "node-fetch";
+import * as core from "@actions/core";
 
 export enum BenchmarkStatus {
-  PENDING = 'PENDING',
-  RUNNING = 'RUNNING',
-  SUCCESS = 'SUCCESS',
-  ERROR = 'ERROR',
-  CANCELED = 'CANCELED',
-  WARNING = 'WARNING',
+  PENDING = "PENDING",
+  RUNNING = "RUNNING",
+  SUCCESS = "SUCCESS",
+  ERROR = "ERROR",
+  CANCELED = "CANCELED",
+  WARNING = "WARNING",
 }
 
-export type UploadRequest = {
-  benchmarkName?: string
-  repoOwner?: string
-  repoName?: string
-  pullRequestId?: string
-  branch?: string,
-  commitSha?: string,
-  env?: { [key: string]: string },
-  agent: string,
-  androidApiLevel?: number,
-  iOSVersion?: number,
-  includeTags: string[],
-  excludeTags: string[],
-  appBinaryId?: string,
-  deviceLocale?: string,
-}
+export type CloudUploadRequest = {
+  benchmarkName?: string;
+  repoOwner?: string;
+  repoName?: string;
+  pullRequestId?: string;
+  branch?: string;
+  commitSha?: string;
+  env?: { [key: string]: string };
+  agent: string;
+  androidApiLevel?: number;
+  iOSVersion?: number;
+  includeTags: string[];
+  excludeTags: string[];
+  appBinaryId?: string;
+  deviceLocale?: string;
+};
+
+type MaestroUploadRequest = Omit<
+  CloudUploadRequest,
+  "benchmarkName" | "repoOwner" | "repoName" | "agent"
```

Comment:
We do have repoOwner, repoName and agent on Robin Request. Also I think maybe comment which request is for robin or new infra

Reply:
I tried - "repoOwner" is giving an error.

Added back "repoName" and "agent".

Reply:
Strange 🤔 What error?

Reply:
Fixed with the changes on copilot. 🙌